### ISSUE TITLE
Native Participate / Commits support

### ISSUE THREAD

**Issue by annevk** [tabatkins/bikeshed#616]
Due to #600 I realized that custom values in the `<dl>` at the start might not be optimal as you cannot control the order. So perhaps I should try to get our current structure supported by bikeshed. See https://notifications.spec.whatwg.org/, https://url.spec.whatwg.org/, and https://dom.spec.whatwg.org/ for examples.

The basic input would be a repository, Twitter account, and optionally a legacy bug link (and even that follows a pattern).

WDYT?

**Comment by annevk** [comment]
https://xhr.spec.whatwg.org/ has an experiment with linking to tests from there, but is not yet converted to bikeshed.

**Comment by tabatkins** [comment]
> Due to #600 I realized that custom values in the <dl> at the start might not be optimal as you cannot control the order.

Are you just referring to the fact that all "custom" things go at the bottom?  As far as I know, I maintain the relative order of custom things.

> The basic input would be a repository, Twitter account, and optionally a legacy bug link (and even that follows a pattern).

How do you feel about just a url+text line?  You'd just specify two of them per spec, once for GitHub and once for Twitter.  That way nobody has to negotiate for changes to the structure.

Oh, and `Test Suite` is an existing metadata. It currently only accepts a URL, but I can change it to be url+text.

**Comment by annevk** [comment]
Yeah, that custom goes at the bottom changing that would also be fine. It's not much of a hassle to maintain some markup that's mostly static anyway. How would a url+text line get me the current result?

**Comment by tabatkins** [comment]
From the specs you've linked to, each of the `<dd>`s is just of the form `<a href=...>...</a>`.  So you'd just write:

```
Commits: GitHub whatwg/notifications/commits https://github.com/whatwg/notifications/commits
Commits: @notifyapi https://twitter.com/notifyapi
```

and that would give you the result you have today.

**Comment by annevk** [comment]
Yeah, Commits is like that, Participate is not though.

**Comment by tabatkins** [comment]
Okay, I'm gonna rejigger things a bit, because handling the divergences in heading content is getting annoying.

First, I'm gonna add a generalized easy metadata to include/omit parts of the heading.  That way I can add things that y'all want and pref them off by default, or maybe pref them *on* by default and omit them from CSS, etc.

Second, I'm gonna add a mechanism for *ordering* the header parts.

Third, I'm gonna rename Feedback to Participate (it feels more inviting, and works with the existing mailing-list entry), add a GitHub line formatted like y'all do (when repo-issue-tracking is turned on), and add an IRC metadata.

I suppose I can add a legacy-issue-tracker metadata too, and figure out what the "issues" part of the Participate needs to look like based on all the stuff.

Then I'll add Commits metadata, formatted like above.

This all sound workable?

**Comment by annevk** [comment]
Sounds fairly reasonable.

**Comment by domenic** [comment]
This bit me today as I tried to add translations but it ended up on top of participate and commits.

**Comment by annevk** [comment]
Note that #1155 makes this somewhat redundant for the WHATWG.
